### PR TITLE
adding logging statements and refactored the docker dev environment

### PR DESCRIPTION
**Pull request by julien51**
# Description

This is not fully ready for prime time but I had a few questions so I wanted to open that PR.
In this state, it "works" though:
* it starts ganache
* deploys unlock
* deploys an ERC20 contract
* mints some of the ERC20
* transfers some Eth from accounts[0] to ETHEREUM_ADDRESS
* deploys an Eth Lock
* deploys and ERC20 lock

The idea is that we would replace our "local" ganache by this docker cluster.
Eventually I think we should also add a db (for locksmith/rover) to the cluster and any "infra" we need to do local dev.

# Checklist:

- [X] 1 PR, 1 purpose: my Pull Request applies to a single purpose
  - [X] This PR only contains configuration changes (package.json, etc.)
  - [ ] This PR only contains code changes (if configuration changes are required, do a separate PR first, then re-base)
- [ ] My code follows the style guidelines of this project, including naming conventions
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If my code adds or changes components, I have written corresponding stories with Storybook
- [ ] I have performed a self-review of my own code
- [ ] If my code involves visual changes, I am adding applicable screenshots to this thread

### CODE REVIEW

**Comment by julien51** [comment]
Note: I removed `docker/development/.env` in favor of `../.env.dev.local` which we already use across the apps.

File: docker/development/deploy-locks.js [hunk 1]
```diff
@@ -45,27 +45,45 @@ async function mintForAddress(
...
-  )
+async function deployLock(web3Service, wallet, account, lock) {
+  let promise = new Promise(resolve => {
+    wallet.once('transaction.new', hash => {
+      let transaction = {
+        hash,
+      }
+      web3Service.on('transaction.updated', (hash, update) => {
+        transaction = { ...transaction, ...update }
+        if (transaction.status == 'mined') {
```
Comment (julien51) [inline docker/development/deploy-locks.js]: This whole dance is so we can log the lock address.

File: docker/docker-compose.development.yml [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
-        - '16686:16686'
-        - '14268:14268'
-        - '9411:9411'
\ No newline at end of file
+  ganache:
+    restart: always
+    build:
+      context: ./development
+      dockerfile: ganache.dockerfile
+    env_file: ../.env.dev.local
```
Comment (akeem) [inline docker/docker-compose.development.yml]: I do think that we should revisit the idea of the using these environment specific .env files https://github.com/motdotla/dotenv#should-i-have-multiple-env-files

File: docker/docker-compose.development.yml [hunk 3]
```diff
@@ -1,21 +1,21 @@
...
+    ports:
+      - 8545:8545
+  jaeger:
+    restart: always
+    image: jaegertracing/all-in-one:1.12
```
Comment (akeem) [inline docker/docker-compose.development.yml]: Was this intended to be a change in image version here?

File: docker/development/deploy-unlock.js [hunk 4]
```diff
@@ -14,12 +14,8 @@ const TokenDeployer = require('./deploy-locks')
...
 const host = process.env.HTTP_PROVIDER_HOST
 const port = process.env.HTTP_PROVIDER_PORT
-let deployedLockAddress
-let purchaser = process.env.PURCHASER_ADDRESS
-let recipientAddress = process.env.ERC20_TOKEN_RECIPIENT
-let bootstrapTransferAmount = process.env.BOOTSTRAP_AMOUNT
-let bootstrapTranferRecipient = process.env.ETHEREUM_ADDRESS
-let contractOwnerAddress = process.env.CONTRACT_OWNER_ADDRESS
+let purchaser = process.env.LOCKSMITH_PURCHASER_ADDRESS // This is the locksmith user account
+let userAdress = process.env.ETHEREUM_ADDRESS // This is a user account
```
Comment (akeem) [inline docker/development/deploy-unlock.js]: Would you mind if we adjusted the spelling here and where this is referenced, also what is meant by user account here?

File: docker/development/deploy-unlock.js [hunk 5]
```diff
@@ -14,12 +14,8 @@ const TokenDeployer = require('./deploy-locks')
  */
 const host = process.env.HTTP_PROVIDER_HOST
 const port = process.env.HTTP_PROVIDER_PORT
-let deployedLockAddress
-let purchaser = process.env.PURCHASER_ADDRESS
-let recipientAddress = process.env.ERC20_TOKEN_RECIPIENT
-let bootstrapTransferAmount = process.env.BOOTSTRAP_AMOUNT
-let bootstrapTranferRecipient = process.env.ETHEREUM_ADDRESS
-let contractOwnerAddress = process.env.CONTRACT_OWNER_ADDRESS
+let purchaser = process.env.LOCKSMITH_PURCHASER_ADDRESS // This is the locksmith user account
```
Comment (akeem) [inline docker/development/deploy-unlock.js]: let's adjust this be something like programmaticPurchaser for more clarity

File: docker/docker-compose.development.yml (hunk 2 above)
Comment (julien51) [reply docker/docker-compose.development.yml]: Well this changes actually makes one less `.env` file..

File: docker/docker-compose.development.yml (hunk 3 above)
Comment (julien51) [reply docker/docker-compose.development.yml]: Oups!

File: docker/development/deploy-unlock.js (hunk 4 above)
Comment (julien51) [reply docker/development/deploy-unlock.js]: What do you call adjust the spelling?

File: docker/development/deploy-unlock.js (hunk 5 above)
Comment (julien51) [reply docker/development/deploy-unlock.js]: Works for me!

**Pull request merged**